pkg/snowflake/task.go

```diff
@@ -0,0 +1,257 @@
+	return tb
+}
+
+// WithComment adds a comment to the TaskBuilder
+func (tb *TaskBuilder) WithComment(c string) *TaskBuilder {
+	tb.comment = c
+	return tb
+}
+
+// WithTimeout adds a timeout to the TaskBuilder
+func (tb *TaskBuilder) WithTimeout(t int) *TaskBuilder {
+	tb.user_task_timeout_ms = t
+	return tb
+}
+
+// WithDepedency adds an after task dependency to the TaskBuilder
+func (tb *TaskBuilder) WithDependency(after string) *TaskBuilder {
+	tb.after = after
+	return tb
+}
+
+// WithCondition adds a when condition to the TaskBuilder
+func (tb *TaskBuilder) WithCondition(when string) *TaskBuilder {
+	tb.when = when
+	return tb
+}
+
+// WithStatement adds a sql statement to the TaskBuilder
+func (tb *TaskBuilder) WithStatement(sql string) *TaskBuilder {
+	tb.sql_statement = sql
+	return tb
+}
+
+// Task returns a pointer to a Builder that abstracts the DDL operations for a task.
+//
+// Supported DDL operations are:
+//   - CREATE TASK
+//   - ALTER TASK
+//   - DROP TASK
+//   - DESCRIBE TASK
+//
+// [Snowflake Reference](https://docs.snowflake.com/en/user-guide/tasks-intro.html#task-ddl)
+func Task(name, db, schema string) *TaskBuilder {
+	return &TaskBuilder{
+		name:   name,
+		db:     db,
+		schema: schema,
+	}
+}
+
+// Create returns the SQL that will create a new task
+func (tb *TaskBuilder) Create() string {
+	q := strings.Builder{}
+	q.WriteString(`CREATE`)
+
+	q.WriteString(fmt.Sprintf(` TASK %v`, tb.QualifiedName()))
+	q.WriteString(fmt.Sprintf(` WAREHOUSE = %v`, tb.warehouse))
+
+	if tb.schedule != "" {
+		q.WriteString(fmt.Sprintf(` SCHEDULE = '%v'`, tb.schedule))
+	}
+
+	if len(tb.session_parameters) > 0 {
+		q.WriteString(fmt.Sprintf(` %v`, strings.Join(tb.session_parameters, ", ")))
+	}
+
+	if tb.comment != "" {
+		q.WriteString(fmt.Sprintf(` COMMENT = '%v'`, tb.comment))
+	}
+
+	if tb.user_task_timeout_ms > 0 {
+		q.WriteString(fmt.Sprintf(` USER_TASK_TIMEOUT_MS = %v`, tb.user_task_timeout_ms))
+	}
+
+	if tb.after != "" {
+		q.WriteString(fmt.Sprintf(` AFTER %v`, tb.GetFullName(tb.after)))
+	}
+
+	if tb.when != "" {
+		q.WriteString(fmt.Sprintf(` WHEN %v`, tb.when))
+	}
+
+	if tb.sql_statement != "" {
+		q.WriteString(fmt.Sprintf(` AS %v`, tb.sql_statement))
+	}
+
+	return q.String()
+}
+
+// getSessionParameters gets the actual parameter
+func getSessionParameters(params []string) []string {
+	out := make([]string, 0)
+	for _, p := range params {
+		s := strings.Split(p, "=")
+		out = append(out, strings.TrimSpace(s[0]))
+	}
+	return out
+}
+
+// ChangeWarehouse returns the sql that will change the warehouse for the task.
+func (tb *TaskBuilder) ChangeWarehouse(newWh string) string {
+	return fmt.Sprintf(`ALTER TASK %v SET WAREHOUSE = %v`, tb.QualifiedName(), newWh)
+}
+
+// ChangeSchedule returns the sql that will change the schedule for the task.
+func (tb *TaskBuilder) ChangeSchedule(newSchedule string) string {
+	return fmt.Sprintf(`ALTER TASK %v SET SCHEDULE = '%v'`, tb.QualifiedName(), newSchedule)
```

Comment:
This (and the rest of the queries here) should probably use escaping - https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/37ea8346c3ac656088a1cc47efc4f602c9966965/pkg/snowflake/escaping.go#L7